File: 卡码/代码随想录/1/07-回溯/Python/01-组合/LeetCode_0077.py
#77. 组合
from typing import *

class Solution:
  def __init__(self):
    self.res:List[List[int]]=[]
    self.path:List[int]=[]
  def backtracking(self,n:int,k:int,index:int):
    if k==len(self.path):
      self.res.append(self.path[:])
      return
    for i in range(index,n+2-(k-len(self.path))):
      self.path.append(i)
      self.backtracking(n,k,i+1)
      self.path.pop()
  def combine(self, n: int, k: int) -> List[List[int]]:
    self.backtracking(n,k,1)
    return self.res

File: 卡码/代码随想录/1/07-回溯/Js/01-组合/LeetCode_0077.js
//77. 组合
const combine = function(n, k) {
  const res=new Array();
  const path=new Array();
  const backtracking=(n,k,index)=>{
    if(k==path.length){
      res.push(Array.from(path));
      return;
    }
    for(let i=index;i<=n+1-(k-path.length);i++){
      path.push(i);
      backtracking(n,k,i+1);
      path.pop();
    }
  }
  backtracking(n,k,1);
  return res;
};